src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,55 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.ValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_AGE = 18;
+    private static final int MINIMAL_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUserDataNull(user);
+        checkLengthOfData(user);
+        checkIsUserAdult(user);
+        isLoginInStorage(user);
+        return storageDao.add(user);
+    }
+
+    private User isLoginInStorage(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new ValidationException("User with this login name already exist");
+        }
+        return user;
+    }
+
+    private User checkIsUserAdult(User user) {
+        if (user.getAge() < MINIMAL_AGE) {
+            throw new ValidationException("Age must equals or more than "
+                    + MINIMAL_AGE);
+        }
+        return user;
+    }
+
+    private User checkLengthOfData(User user) {
```

Comment:
shouldn't this method be `void`?

and I would better call this method like `verifyCorrectCredentials`